src/test/app/Offer_test.cpp

```diff
@@ -795,22 +850,831 @@ class Offer_test : public beast::unit_test::suite
             owners ("eve", 1));
     }
 
+    void
+    testcaseSelfCross()
+    {
+        testcase ("Self-crossing");
+
+        using namespace jtx;
+
+        auto const gw = Account {"gateway"};
+        auto const USD = gw["USD"];
+        auto const BTC = gw["BTC"];
+
+        Env env {*this};
+        env.fund (XRP (10000), gw);
+        env (offer (gw, BTC (500), USD (100)));
+        env (offer (gw, USD (100), BTC (500)));
+        env.close ();
```

Comment:
Should work great.  Too many things change to check them all every time.  I'm sure you'll make a good best guess about what to look at.